drivers/hsc/hsc.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * 
+ */
+
+/**
+ * @ingroup     drivers_hsc
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Honeywell HSC temperature and pressure sensor.
+ *
+ * @author      Quang Pham
+ *
+ * @}
+ */
+
+#include "math.h"
+#include "hsc.h"
+#include "hsc_internals.h"
+#include "hsc_params.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define I2C_DEVNUM          (1)
+#define DEV_I2C      (dev->params.i2c_dev)
+#define DEV_ADDR     (dev->params.i2c_addr)
+
+// Internal function prototypes
+// Reading raw (bit) value.  For faster reading, can be removed
+static int _read_ut(const hsc_t *dev, int32_t *ut);
+static int _read_up(const hsc_t *dev, int32_t *up);
+
+/*---------------------------------------------------------------------------*
+ *                          HSC Core API                                     *
+ *---------------------------------------------------------------------------*/
+
+int hsc_init(hsc_t *dev, const hsc_params_t *params)
+{
+    dev->params = *params;
+    uint8_t buf[HSC_FULL_DATA_LENGTH*I2C_DEVNUM] = {0};
```

Comment:
Why multiply by `I2C_DEVNUM`?

Reply:
I2C_DEVNUM will be used in other function for parallel access. I remove for now for lighter operation.